### ISSUE TITLE
Editing Volunteer Opportunities causes 500 error

### ISSUE THREAD

**Issue by burlexpo** [bethlakshmi/GBE2#614]
When user clicks 'edit' on legitimate Volunteer Opportunity events that are listed on the live site (http://burlesque-expo.com/scheduler/list), result is *Server Error (500)*

It happens consistently on Events (the "scheduleable" type) that are Volunteer Opportunities.  "Special Event" and "Rehearsal Slot" do not have this problem.  

We need to identify whether this is a recent regression or some gook left from last year.

-------- Forwarded Message --------
Subject: 	[Django] ERROR (EXTERNAL IP): Internal Server Error: /scheduler/edit/GenericEvent/1434
Date: 	Fri, 18 Dec 2015 02:31:48 -0000
From: 	webdev@burlesque-expo.com
To: 	webdev@burlesque-expo.com, info@burlesque-expo.com

Traceback (most recent call last):

  File "/home/gbelive/webapps/gbelive/lib/python2.7/django/core/handlers/base.py", line 112, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)

  File "/home/gbelive/webapps/gbelive/expo/scheduler/views.py", line 362, in edit_event
    return edit_event_display(request, item)

  File "/home/gbelive/webapps/gbelive/expo/scheduler/views.py", line 418, in edit_event_display
    context.update(show_potential_workers(item))

  File "/home/gbelive/webapps/gbelive/expo/scheduler/views.py", line 559, in show_potential_workers
    available in eval(v.availability)])]

TypeError: 'function' object has no attribute '__getitem__'

<WSGIRequest
path:/scheduler/edit/GenericEvent/1434,
GET:<QueryDict: {}>,
POST:<QueryDict: {}>,
COOKIES:{'csrftoken': 'ZRh4ZtvbRlRcBPGpvZSfAm7CVIKamlBm',
 'django_language': 'en-us',
 'sessionid': 'q0yxd0trjhf9wj2qjztdb47xegh22g2w'},
META:{u'CSRF_COOKIE': u'ZRh4ZtvbRlRcBPGpvZSfAm7CVIKamlBm',
 'DOCUMENT_ROOT': '/usr/local/apache2/htdocs',
 'GATEWAY_INTERFACE': 'CGI/1.1',
 'HTTP_ACCEPT': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8',
 'HTTP_ACCEPT_ENCODING': 'gzip, deflate, sdch',
 'HTTP_ACCEPT_LANGUAGE': 'en-US,en;q=0.8',
 'HTTP_CONNECTION': 'close',
 'HTTP_COOKIE': 'django_language=en-us; sessionid=q0yxd0trjhf9wj2qjztdb47xegh22g2w; csrftoken=ZRh4ZtvbRlRcBPGpvZSfAm7CVIKamlBm',
 'HTTP_FORWARDED_REQUEST_URI': '/scheduler/edit/GenericEvent/1434',
 'HTTP_HOST': 'burlesque-expo.com',
 'HTTP_HTTPS': 'off',
 'HTTP_HTTP_X_FORWARDED_PROTO': 'http',
 'HTTP_REFERER': 'http://burlesque-expo.com/scheduler/list',
 'HTTP_UPGRADE_INSECURE_REQUESTS': '1',
 'HTTP_USER_AGENT': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/47.0.2526.80 Safari/537.36',
 'HTTP_X_FORWARDED_FOR': '108.20.2.124',
 'HTTP_X_FORWARDED_HOST': 'burlesque-expo.com',
 'HTTP_X_FORWARDED_PROTO': 'http',
 'HTTP_X_FORWARDED_SERVER': 'burlesque-expo.com',
 'HTTP_X_FORWARDED_SSL': 'off',
 'PATH_INFO': u'/scheduler/edit/GenericEvent/1434',
 'PATH_TRANSLATED': '/home/gbelive/webapps/gbelive/expo/expo/wsgi.py/scheduler/edit/GenericEvent/1434',
 'QUERY_STRING': '',
 'REMOTE_ADDR': '127.0.0.1',
 'REMOTE_PORT': '40551',
 'REQUEST_METHOD': 'GET',
 'REQUEST_URI': '/scheduler/edit/GenericEvent/1434',
 'SCRIPT_FILENAME': '/home/gbelive/webapps/gbelive/expo/expo/wsgi.py',
 'SCRIPT_NAME': u'',
 'SERVER_ADDR': '127.0.0.1',
 'SERVER_ADMIN': '[no address given]',
 'SERVER_NAME': 'burlesque-expo.com',
 'SERVER_PORT': '80',
 'SERVER_PROTOCOL': 'HTTP/1.0',
 'SERVER_SIGNATURE': '',
 'SERVER_SOFTWARE': 'Apache/2.2.25 (Unix) mod_wsgi/3.5 Python/2.7.9',
 'mod_wsgi.application_group': 'web489.webfaction.com|',
 'mod_wsgi.callable_object': 'application',
 'mod_wsgi.enable_sendfile': '0',
 'mod_wsgi.handler_script': '',
 'mod_wsgi.input_chunked': '0',
 'mod_wsgi.listener_host': '',
 'mod_wsgi.listener_port': '14983',
 'mod_wsgi.process_group': 'gbelive',
 'mod_wsgi.queue_start': '1450405908030161',
 'mod_wsgi.request_handler': 'wsgi-script',
 'mod_wsgi.script_reloading': '1',
 'mod_wsgi.version': (3, 5),
 'wsgi.errors': <mod_wsgi.Log object at 0x7f890501e2b0>,
 'wsgi.file_wrapper': <built-in method file_wrapper of mod_wsgi.Adapter object at 0x7f8904c7cbe8>,
 'wsgi.input': <mod_wsgi.Input object at 0x7f890501e570>,
 'wsgi.multiprocess': True,
 'wsgi.multithread': True,
 'wsgi.run_once': False,
 'wsgi.url_scheme': 'http',
 'wsgi.version': (1, 0)}>

**Comment by jonkiparsky** [comment]
This is cleared up by #488. 
@bethlakshmi - any thoughts on when we can get the latest commits onto test?

**Comment by bethlakshmi** [comment]
I'm glad to have that happen whenever. 
Am I still the only person who can update on test?? the procedure is seriously not hard and it's documented on the wiki, but it's this:

cd ~/git-drop
rm -rf GBE2.old
mv GBE2 GBE2.old
git clone https://github.com/bethlakshmi/GBE2.git 
cp -r ~/git-drop/GBE2/expo/ ~/webapps/gbe2016test/
cd ~/webapps/gbe2016test/expo
python manage.py collectstatic
python manage.py migrate gbe

And I believe that's what's in the wiki.

**Comment by bethlakshmi** [comment]
Of note, though, there's something not working in the data - 

I just ran into a kink of this problem - the current gbe live volunteer windows are booked for the old dates from 2015, and not the new dates for 2016.  I don't think our migrations are covering this properly, before we push #488 to live, we've got to find a way to uncorrupt the data.

Example:

http://gbelive.webfactional.com/scheduler/edit/GenericEvent/1469
Is a special event.

It has a volunteer event that shows up on the form above as "Saturday", "10:00PM" for 2 hours.

If you go to that volunteer event and view the detail:

http://gbelive.webfactional.com/scheduler/details/404

Note that the date is Feb 21, 2015, which I will bet money is a ... Saturday!!! 
From last year.

The static dates on Volunteer Opportunity hard coding where not updated and now the data is wrong.
If we fix the data, we'll be OK - but the migration itself isn't taking care of it - to test it, grab a copy of the live data and test locally... I'm seeing this in my local vagrant box as I work on the volunteer bid acceptance stuff.
